CondorMonitoring/JobCounter.py

```diff
@@ -589,72 +604,99 @@ def main():
-                    jType = 'Reco'
-                elif jType not in jobTypes:  # If job type is not standard
-                    jType = jobType(jobId, schedd_name, task)
-
-                siteRunning = job.get('MATCH_EXP_JOBGLIDEIN_CMSSite', '')
-                if siteName(siteRunning):  # If job is currently running
-                    siteToExtract = [siteRunning]
-
-                # Ignore jobs to the T1s from the Tier-0 pool
-                # Avoid double accounting with the global pool
-                if collector_name == tier0_pool[0]:
-                    if not 'T2_CH_CERN_T0' in siteToExtract:
-                        continue
+                    jobId = str(job['ClusterID']) + '.' + str(job['ProcId'])
+                    status = int(job['JobStatus'])
 
-                # Ignore jobs to T2_CH_CERN_T0 from the global pool
-                # Avoid double accounting with the Tier-0 pool
-                if collector_name == global_pool[0]:
-                    if 'T2_CH_CERN_T0' in siteToExtract:
+                    if jobKeys[schedd_type]['taskname'] not in job:
+                        # CRAB has a lot of these which are not tagged and they are task processes.
+                        # They should not be accounted for anything.
+                        #print 'I found a job coming from %s, but it does not have needed classads. JobID %s' % (schedd_type, jobId)
+                        continue
+                    workflow = None
+                    task = None
+                    if schedd_type == 'prodschedd':
+                        workflow = job['WMAgent_SubTaskName'].split('/')[1]
+                        task = job['WMAgent_SubTaskName'].split('/')[-1]
+                        jType = job['CMS_JobType']
+                    elif schedd_type == 'crabschedd':
+                        workflow = re.sub('[:]', '', job['CRAB_UserHN'])
+                        task = job['CRAB_ReqName']
+                        jType = 'Crab3'
+                        if job['TaskType'] == 'ROOT':
+                            # Means it is a ROOT task, which is running only on scheduler.
+                            # and it should not be accounted.
+                            continue
+                    else:
+                        print 'How did you got here?! Are you developing something?'
+                        raise
+                    try:  # it can be an ExprTree
+                        cpus = get_int(job['RequestCpus'])
+                    except:
+                        # Catch any except in case something in the future would change in HTCondor or how ExprTree is evaluated.
+                        # It is not correct to assume it is 1 cpu. Skip this job.
+                        print 'Failed to extract RequestCpus from this job %s' % job
+                        continue
+                    siteToExtract = str(job['DESIRED_Sites']).replace(' ', '').split(",")
+                    while '' in siteToExtract:
```

Comment:
I don't really like it :) Why da hell crab is adding empty strings in the desired sites?
Anyways, I suggest to replace these 3 lines by a list comprehension like
`siteToExtract = [site for site in job['DESIRED_Sites'].replace(' ', '').split(",") if site]`